### PR TITLE
feat(github): add flipt tag driven protobuf generate action

### PR DESCRIPTION
**Pull request by GeorgeMac**
I am unsure if this will work. However, I thought it would be fun to try.

The idea is that the Flipt release tag can be provided to this action, and it will take care of generating the client directly into a PR for us.

### CODE REVIEW

File: .github/workflows/proto-upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+on:
+  workflow_dispatch:
```
Comment (markphelps) [inline .github/workflows/proto-upgrade.yml:2]: if this works we could use the `release` trigger to have this automatically run after Flipt release is created

**Review by markphelps — APPROVED** [review]

File: .github/workflows/proto-upgrade.yml (hunk 1 above)
Comment (GeorgeMac) [reply .github/workflows/proto-upgrade.yml:2]: I like that idea. Does is work cross-repo? I remember having issues with other cross repo-actions.

In the end, we created a github app and used an action to use the app credentials to create an access token which we then used to trigger an action on another repo. Was a rube goldberg machine, but it worked.

File: .github/workflows/proto-upgrade.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Setup go-task/task
+        uses: 'lagren/setup-task@v1'
+
+      - name: Clone Flipt into sub-directory
+        run: gh repo clone flipt-io/flipt
```
[the remark this answers is not in the document]
Comment (GeorgeMac) [reply .github/workflows/proto-upgrade.yml:30]: Yeah a depth would be good. Unsure yet how I can just partially clone a dir. I would love to just get the rpc directory.

Though I might also need _tools and scripts too.

File: .github/workflows/proto-upgrade.yml (hunk 1 above)
Comment (markphelps) [reply .github/workflows/proto-upgrade.yml:2]: yeah, you cant use the provided `secrets.GITHUB_TOKEN` when doing triggers on release from other repos unfortunately, or even your same repo I think. something about preventing infinite loops. but we can use a PAT as an org secret. I already have one that can create releases

Comment (markphelps) [reply .github/workflows/proto-upgrade.yml:2]: also i think its worth keeping the `workflow_dispatch` trigger around too like you have it, so we can manually release if necessary

File: .github/workflows/proto-upgrade.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Clone Flipt into sub-directory
+        run: gh repo clone flipt-io/flipt
+        env:
+          GH_TOKEN: ${{ github.token }}
```
Comment (markphelps) [inline .github/workflows/proto-upgrade.yml:28]: I think its `secrets.GITHUB_TOKEN`

Comment (GeorgeMac) [reply .github/workflows/proto-upgrade.yml:28]: This was actually suggested in an error by github. Is it a new or outdated thing perhaps?

Comment (markphelps) [reply .github/workflows/proto-upgrade.yml:28]: weird, never seen that https://docs.github.com/en/actions/security-guides/automatic-token-authentication

**Pull request merged**